tests/integrationv2/test_npn.py

```diff
@@ -0,0 +1,204 @@
+import pytest
+import copy
+
+from configuration import available_ports, ALL_TEST_CIPHERS, ALL_TEST_CURVES, MINIMAL_TEST_CERTS, PROTOCOLS
+from common import ProviderOptions, Protocols
+from fixtures import managed_process
+from providers import OpenSSL, S2N, Provider
+from utils import invalid_test_parameters, get_parameter_name, to_bytes
+
+
+# NPN not supported in TLS1.3
+TLS_PROTOCOLS = [x for x in PROTOCOLS if x.value < Protocols.TLS13.value]
+
+# Output indicating NPN status
+S2N_NPN_MARKER = "WITH_NPN"
+S2N_APPLICATION_MARKER = "Application protocol: "
+OPENSSL_SERVER_NPN_MARKER = "NEXTPROTO is "
+# The OpenSSL client uses "(1)" to indicate that a protocol was selected from
+# the server's advertised list. "(2)" indicates the client couldn't find any overlap
+# with the server's list and had to select from its own list.
+OPENSSL_CLIENT_NPN_MARKER = "Next protocol: (1) "
+OPENSSL_CLIENT_NPN_NO_OVERLAP_MARKER = "Next protocol: (2) "
+
+# Test lists
+PROTOCOL_LIST = 'http/1.1,h2,h3'
+PROTOCOL_LIST_ALT_ORDER = 'h2,h3,http/1.1'
+PROTOCOL_LIST_NO_OVERLAP = 'spdy'
+
+def s2n_client_npn_handshake(managed_process, cipher, curve, certificate, protocol, provider, server_list):
+    options = ProviderOptions(
+        port=next(available_ports),
+        cipher=cipher,
+        curve=curve,
+        key=certificate.key,
+        cert=certificate.cert,
+        protocol=protocol,
+        insecure=True,
+    )
+
+    client_options = copy.copy(options)
+    client_options.mode = Provider.ClientMode
+    # Flags to turn on NPN for s2nc
+    client_options.extra_flags = ['--alpn', PROTOCOL_LIST, '--npn']
+
+    server_options = copy.copy(options)
+    server_options.mode = Provider.ServerMode
+    # Flags to turn on NPN for OpenSSL server
+    server_options.extra_flags = ['-nextprotoneg', server_list]
+
+    server = managed_process(provider, server_options, timeout=5)
```

Comment:
should this be `s2n_server` for consistency?

Reply:
Nope, it's an Openssl server.